dm/dm/master/server.go

```diff
@@ -1682,7 +1678,10 @@ func (s *Server) waitOperationOk(
 				}
 			case *pb.StartTaskRequest, *pb.UpdateTaskRequest, *pb.OperateTaskRequest:
 				if expect == pb.Stage_Stopped && len(queryResp.SubTaskStatus) == 0 {
-					return true, "", queryResp, nil
+					workerQueryResponse = s.checkWorkerStatus(sourceID, cli)
```

Comment:
i am not sure if this fix https://github.com/pingcap/ticdc/issues/3254, could you mock `task has an error` in `TestWaitOperationOkRightResult` to verify this patch works fine?